src/Concordium/Client/Runner.hs

```diff
@@ -3445,81 +3448,391 @@ processIdentityShowCmd action backend =
+        printSuccess
+    UnbanNode nodeIp ->
+      withClient backend $
+        unbanPeerV2 (IpAddress nodeIp) >>=
+        printSuccess
+    GetAncestors amount block ->
+      withClient backend $ do
+        b <- readOrFailM Best block
+        getAncestorsV2 b (fromIntegral amount) >>=
+          printResponseValueAsJSON'
+    GetBranches ->
+      withClient backend $
+        getBranchesV2  >>=
+        printResponseValueAsJSON'
+    GetBannedPeers ->
+      withClient backend $
+        getBannedPeersV2 >>=
+        getResponseValueOrFail'' >>=
+        -- The coercion is done to avoid either an orphan
+        -- instance here or polluting `Concordium.GRPC2`.
+        printJSONValues . toJSON . fmap (coerce :: IpAddress -> Text)
+    Shutdown ->
+      withClient backend $
+        shutdownV2 >>=
+        printSuccess
+    DumpStart file raw ->
+      withClient backend $
+        dumpStartV2 file raw >>=
+        printSuccess
+    DumpStop ->
+      withClient backend $
+        dumpStopV2 >>=
+        printSuccess
+    GetIdentityProviders block ->
+      withClient backend $
+        readOrFailM Best block >>=
+        getIdentityProvidersV2 >>=
+        printResponseValueAsJSON'
+    GetAnonymityRevokers block ->
+      withClient backend $
+        readOrFailM Best block >>=
+        getAnonymityRevokersV2 >>=
+        printResponseValueAsJSON'
+    GetCryptographicParameters block ->
+      withClient backend $
+        readOrFailM Best block >>=
+        getCryptographicParametersV2 >>=
+        printResponseValueAsJSON'
   where
-    printSuccess (Left x)  = liftIO . putStrLn $ x
-    printSuccess (Right x) = liftIO $ if x then putStrLn "OK" else putStrLn "FAIL"
-    printSuccess' = printSuccess . fmap grpcResponseVal
-
--- |Look up block infos all the way to genesis.
-loop :: Either String Value -> ClientMonad IO ()
-loop v =
-  case v of
-    Left err       -> liftIO $ putStrLn err
-    Right (AE.Object m) ->
-      case KM.lookup "blockParent" m of
-        Just (String parent) -> do
-          printJSON v
-          case KM.lookup "blockSlot" m of
-            Just (AE.Number x) | x > 0 ->
-              fmap grpcResponseVal <$> getBlockInfo parent >>= loop
-            _ -> return () -- Genesis block reached.
-        _ -> error "Unexpected return value for block parent."
-    _ -> error "Unexptected return value for getBlockInfo."
-
+    -- |Print the response value under the provided mapping,
+    -- or fail if the response contained an error.
+    printResponseValue f res =
+      getResponseValueOrFail' f res >>= liftIO . print
+
+    -- |Print the response value under the provided mapping
+    -- as JSON, or fail if the response contained an error.
+    printResponseValueAsJSON res f = do
+      v <- getResponseValueOrFail' f res
+      printJSONValues . toJSON $ v
+
+    -- |Print the response value as JSON, or fail if the response
+    -- contained an error.
+    printResponseValueAsJSON' res =
```

Comment:
Seems like you don't use `printResponseValueAsJSON` directly. So perhaps that is the solution.
But otherwise, I would have renamed the non-prime version to something with `map`.